text/0478-tracked-properties-updates.md

```diff
@@ -0,0 +1,1130 @@
+only need to use them for plain, undecorated properties on POJOs, and for Ember
+proxies.
+
+## Detailed design
+
+### Autotracking Interop
+
+Not all of the interop in the original RFC was problematic, and as such, the
+following parts will remain the same:
+
+- `get` - will still autotrack any value that is accessed
+- `set`/`notifyPropertyChange` - will still invalidate any value that was
+  accessed with `get`
+- Computed properties - will still autotrack _if accessed in an autotracking
+  context_.
+
+What will change is that computed properties will no longer autotrack as they
+are being evaluated. Instead, they will follow the same rules as they do
+currently, listing explicit dependencies and only invalidating when one of those
+dependencies changes. Any autotracking that may have occured during the
+computation of the computed will instead no-op, making the computed a _black
+box_.
+
+Since computed properties no longer autotrack, they will need a different
+interop story for tracked properties and autotracking. Autotracking is a very
+general tool - as we saw in the motivation, it's possible to track through
+_function calls_, something that wasn't possible before.
+
+However, we don't need to enable interop with _all_ of autotracking. All we
+really need is the ability to depend on the autotracking equivalents of what
+computed properties already are capable of depending on, so that users can
+convert existing code to autotracking incrementally. Computed properties can
+already depend on:
+
+- Properties
+- Other Computed Properties
+
+The equivalents to these in autotracking are:
+
+- Tracked Properties
+- Native Getters
+
+#### Depending on Tracked Properties
+
+Tracked properties are already instrumented under the hood. They have a native
+setter that calls `notifyPropertyChange`, and this will automatically invalidate
+any computeds that specify the property as a dependency. There is no need to do
+any further work.
+
+#### Depending on Native Getters
+
+Native getters are just that - native. They don't have any special autotracking
+behavior, which was part of the benefits of tracked properties. However, this
+means there is nothing to notify computed properties of changes.
+
+To solve this problem, we propose the `@watchable` decorator. This decorator
+would instrument a native getter with its own autotracking frame, which would
+allow it to track any events in its evaluation. It would coalesce these into its
+own tag, which computed properties (and observers) would be able to depend on:
+
+```js
+import { tracked, watchable } from '@glimmer/tracking';
+
+class Person {
+  @tracked firstName;
+  @tracked lastName;
+
+  @watchable
+  get fullName() {
+    return `${this.firstName} ${this.lastName}`;
+  }
+}
+
+const Profile = EmberObject.extend({
+  // provided on create
+  person: null,
+
+  username: alias('person.fullName'),
+});
+```
+
+`@watchable` would be imported from `@glimmer/tracking` alongside `@tracked`.
```

Comment:
Seems odd to import from `@glimmer` since its only about interop with Ember features (computed and observers).